src/js/background/context-menu.js

```diff
@@ -153,15 +172,42 @@ const relayContextMenus = {
         reuseAliasMenuIdPrefix,
         ""
       );
-
       // Get stored alias data
-      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      const relayAddresses = await relayContextMenus.utils.getAliases();
 
       // Select the correct alias from the stored alias data
       const selectedAliasObject = relayAddresses.filter((alias) => {
         return alias.id === parseInt(selectedAliasId, 10);
       });
 
+      const serverStoragePref = await getCachedServerStoragePref();
+
+    
+      const currentMaskType = selectedAliasObject[0].mask_type;
+      const currentUsedOnValue = selectedAliasObject[0].used_on;
+    
+      const currentPage = new URL(tab.url);
+      const currentPageHostName = currentPage.hostname;
+ 
+      // If the used_on field is blank, then just set it to the current page/hostname. Otherwise, add/check if domain exists in the field
+      const used_on = (currentUsedOnValue === null || currentUsedOnValue === "" ||  currentUsedOnValue === undefined)
+        ? `${currentPageHostName},` : relayContextMenus.utils.addUsedOnDomain(currentUsedOnValue, currentPageHostName);
+
+      // Update server info with site usage
+      const data = {used_on};
+      const options = {
+        auth:true,
+        mask_type: currentMaskType,
+      };
+
+      // Save what site this mask was used on before filling input
+      if (serverStoragePref) {
+        await patchMaskInfo("PATCH", parseInt(selectedAliasId, 10), data, options);
+      } else {
+        selectedAliasObject[0].used_on = used_on;
+        browser.storage.local.set({ relayAddresses: relayAddresses });
```

Comment:
I might be missing a bit too much context in the PR review view, but `relayAddresses` doesn't seem to get changed? Are you sure there's a need here to save it to local storage again?

Reply:
The `selectedAliasObject[0].used_on` is modified, which is part of `relayAddresses` object. We're making sure to save the new `used_on` entry for the selected/used mask. 

I confirmed this by outputting `relayAddresses` before and after the `used_on` setting on L207.

Reply:
I'll add a comment to make this more visible!